lib/puppet/http/client.rb

```diff
@@ -19,17 +19,16 @@ def create_session
 
   def connect(uri, options: {}, &block)
     start = Time.now
+    verifier = nil
+    connected = false
 
-    ssl_context = options.fetch(:ssl_context, nil)
-    include_system_store = options.fetch(:include_system_store, false)
-    ctx = resolve_ssl_context(ssl_context, include_system_store)
     site = Puppet::Network::HTTP::Site.from_uri(uri)
-    verifier = if site.use_ssl?
-                 Puppet::SSL::Verifier.new(site.host, ctx)
-               else
-                 nil
-               end
-    connected = false
+    if site.use_ssl?
```

Comment:
Should we add a test to make sure we don't try to resolve the ssl context for HTTP connections?

Reply:
👍

Reply:
Added https://github.com/puppetlabs/puppet/pull/8051/files#diff-2d59621843d0ea9897e0ef2f860a8e2eR78